gum/gumdarwinmodule.c

```diff
@@ -1896,7 +1897,19 @@ const gchar *
 gum_darwin_module_get_dependency_by_ordinal (GumDarwinModule * self,
                                              gint ordinal)
 {
-  gint i = ordinal - 1;
+  gint i;
+
+  switch (ordinal)
+  {
+    case GUM_BIND_SPECIAL_DYLIB_SELF:
+      return self->name;
+    case GUM_BIND_SPECIAL_DYLIB_MAIN_EXECUTABLE:
+    case GUM_BIND_SPECIAL_DYLIB_FLAT_LOOKUP:
+    case GUM_BIND_SPECIAL_DYLIB_WEAK_LOOKUP:
+      return NULL;
+  }
+
+  i = ordinal - 1;
 
   if (!gum_darwin_module_ensure_image_loaded (self, NULL))
     return NULL;
```

Comment:
We should move this first, so `self->name` is initialized.

Reply:
oops!